presto-common/src/main/java/com/facebook/presto/common/type/TypeSignatureParameter.java

```diff
@@ -13,19 +13,25 @@
  */
 package com.facebook.presto.common.type;
 
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.facebook.drift.annotations.ThriftStruct;
 import com.facebook.presto.common.type.BigintEnumType.LongEnumMap;
 import com.facebook.presto.common.type.VarcharEnumType.VarcharEnumMap;
 
 import java.util.Objects;
 import java.util.Optional;
 
+import static com.facebook.presto.common.type.TypeSignatureParameterUnion.convertToTypeSignatureParameterUnion;
+import static com.facebook.presto.common.type.TypeSignatureParameterUnion.convertToValue;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
+@ThriftStruct
 public class TypeSignatureParameter
 {
     private final ParameterKind kind;
-    private final Object value;
+    private final TypeSignatureParameterUnion value;
```

Comment:
does this work with the previous JSON serde?

Reply:
Yes; JSON uses the `toString()` method, so I simply cast the `TypeSignatureParameterUnion` back to an `Object` in that method.